### DISCUSSION TITLE
Multi Firm web portal

### DISCUSSION THREAD

**Discussion by ecxs** [Budibase/budibase#9027] [Help]
Hi there,

I'm thinking of making a multi firm web portal for the customers of my customer. The idea is, that customers can access the same app, but can only see/access their own data. I was thinking of adding an extra column Company_Id to the Users table, and then filter the tables on the Company_Id. Is this a good/safe approach? I absolutely want to avoid that customers can see other customers data!

Thanks in advance, and best regards, Dick

**Comment by melohagan** [comment]
Hey @ecxs 

Budibase isn't really built with multi-tenancy in mind: https://github.com/Budibase/budibase/discussions/3866#discussioncomment-1863447

I would be careful about relying on a filtered Users table column. Currently there is nothing at the data level to prevent the public API being used to access data from the users table. Also admin/developers could see all the entries in the users table.
